### ISSUE TITLE
varnish 5

### ISSUE THREAD

**Issue by gboddin** [iuscommunity/wishlist#98]
### What package do you want?

varnish 5.0

### Why?

https://github.com/gboddin/edge-repo-dev/blob/master/SPECS/varnish.spec (they have it upstream) is already done, I was wondering if we could merge edge-repo and get rid of it.
 

### OS versions

I would like (to maintain) this package for both RHEL 6 and 7.

### Testing

I agree to test the new package to ensure that it works as expected, we use it production

**Comment by carlwgeorge** [comment]
Howdy @gboddin,

Thanks for opening this request with us.  I have a few points of observation I wanted to write down as we evaluate this.

* IUS normally provides newer versions of stock packages.  Varnish is not available in the stock RHEL repos, but it is available from EPEL (2.1.5 in the EL6 branch, 4.0.3 in the EL7 branch).  We have made exceptions in the past to provide newer versions of EPEL packages, but only after effort has been made to request a rebase from the EPEL maintainer.  EPEL only allows major version changes when the software is considered backwards compatible (mainly in the context of configuration files).  How compatible would an upgrade from varnish 2->5 or 4->5 be?

* Varnish provides their own yum repositories.  However, those repositories do not yet include varnish 5.  The [installation page](https://varnish-cache.org/releases/install_redhat.html) says that "Fresh releases, such as Varnish Cache 5, will be made available soon".  Do you happen to have any insight as to why they haven't done that yet?  If they were ship varnish 5 packages soon, would that be a preferable solution?

**Comment by gboddin** [comment]
Upgrading varnish from  2.0 or 4.0 to 4.1 or 5 will be a complete disaster if the user is not prepared, the vcls won't be compatible and need rewriting.

There's however quite a lot of new features that makes the switch worth it ( background backend refresh, http2, ...). Most of the new features and breaking changes are introduced in 4.1

Varnish seems to only release packages for versions they support commercially ( right now it's version 4.1 ) even if they release other versions. I had a talk with one of the sales reps on the phone who confirmed they're actually using 4.1 with their customers, we could decide to follow version 4.1 (not totally inline with upstream, but close enough) if we want to be perfectly safe and allow people to get commercial support.

The goal was to have a complete varnish suite up to date ( with its vmods ), if EPEL doesn't upgrade varnish to 4.1 we'll be fine.

**Comment by carlwgeorge** [comment]
I started looking around on the varnish repo webserver, and I found [this directory](https://repo.varnish-cache.org/pkg/5.0.0/).  There is no repodata to configure it as a yum repository, but it could be a staging repository, which may mean that a release of varnish 5.0 RPMs could be imminent.  I'm going to do a bit more research and see if I can find out more.

**Comment by ingvarha** [comment]
Hi there

I'm the maintainer of varnish in Fedora and EPEL. While we wait for upstream packaging of varnish-5.0 for redhat and clones/derivates, I may inform you that I ensure my Fedora packages always can be rebuilt for epel5, epel6, and epel7. So if you want a varnish-5.0 package for el7, consider to base that off the Fedora package.

Prebuilt packages here: 
https://copr.fedorainfracloud.org/coprs/ingvar/varnish50/ 
https://copr.fedorainfracloud.org/coprs/ingvar/varnish41/

Sidenote: Rumors says the varnish-cache.com people will eventually provide upstream packages of varnish for most distributions. It's just not on top of their high priority list.

Ingvar

**Comment by carlwgeorge** [comment]
@ingvarha Thanks for reaching out.  IUS does typically fork our spec files from their Fedora counterparts, so I'm with you there.  I also observed what you mentioned about the Fedora spec rebuilding correctly against RHEL releases.  The only that I noticed we would have to do is [rename the package to varnish50u](https://ius.io/Philosophy/#naming-convention), replace most instances of `%{name}` with just `varnish`, and [add `Provides` and `Conflicts` for the stock name](https://ius.io/SafeRepo/#safe-replacement-package).  However, since you already maintain coprs for this, that is probably a preferable solution.

@gboddin I reached out to the varnish folks in their IRC channel.  They believed that their won't be any 5.0 packages in the upstream repo until that release is declared "stable".  They pointed out that 5.0 is labeled as "fresh" on the [releases page](https://www.varnish-cache.org/releases/index.html), which I was told means "not production ready".  IUS stands for "Inline with Upstream Stable".  Considering that the upstream doesn't consider 5.0 stable yet, we are not going to proceed with this request at this time.  Luckily @ingvarha maintains coprs that are compatibile with EL5/6/7, as he linked above.  They include both varnish and varnish-modules.

**Comment by gboddin** [comment]
Ok, thanks @ingvarha @carlwgeorge ! Let's wait a bit then :)

Also thanks IUS for the various packaging tips !

In the end edge will become a private repo since we need some additional patches and IUS is already there.

We'll follow IUS and propose pull requests if we improve anything that would be useful for you,
